asab/web/tenant/providers/zookeeper.py

```diff
@@ -0,0 +1,70 @@
+import typing
+import logging
+
+from .abc import TenantProviderABC
+from ....zookeeper import ZooKeeperContainer
+
+
+L = logging.getLogger(__name__)
+
+
+class ZookeeperTenantProvider(TenantProviderABC):
+	def __init__(self, app, tenant_service, config):
+		super().__init__(app, tenant_service, config)
+		self.Tenants: typing.Set[str] = set()
+
+		# Initialize ZooKeeper Client
+		self.ZooKeeperService = self.App.get_service("asab.ZooKeeperService")
+		self.ZKPath = self.Config.get("zk_path")
+		self.ZookeeperContainer = ZooKeeperContainer(
+			self.ZooKeeperService,
+			config_section_name="zookeeper",
+			z_path=self.ZKPath
+		)
+		self.ZookeeperClient = self.ZookeeperContainer.ZooKeeper
+
+
+	async def get_tenants(self) -> typing.Set[str]:
+		return self.Tenants
+
+
+	async def is_tenant_known(self, tenant: str) -> bool:
+		return tenant in self.Tenants
+
+
+	async def update(self):
+		try:
+			zk_node_exists = await self.ZookeeperClient.exists(self.ZKPath)
+
+			if zk_node_exists:
+				external_tenants = await self.ZookeeperClient.get_children(self.ZKPath)
+				self._set_ready(True)  # Provider was checked => True
+
+			elif zk_node_exists is None:
+				L.warning(
```

Comment:
this still feels a bit inconsistent to me.

when there are no tenants in ZK we should either 
- treat it as an error and do `_set_ready(False)` 
- or treat is as an expected thing and set `self.Tenants` to empty set. (this is probably correct)